subsys/nrf_security/src/drivers/cracen/cracenpsa/src/wpa3_sae.c

```diff
@@ -0,0 +1,1039 @@
+				      const uint8_t *peer_id, size_t peer_id_len)
+{
+	if (peer_id_len != CRACEN_WPA3_SAE_STA_ID_LEN) {
+		return PSA_ERROR_INVALID_ARGUMENT;
+	}
+
+	/** This check can only be done here since peer is required to be set after the user.
+	 *  Oberon verifies that prior to call driver wrapper.
+	 */
+	if (memcmp(peer_id, operation->max_id, peer_id_len) > 0) {
+		memcpy(operation->min_id, operation->max_id, peer_id_len);
+		memcpy(operation->max_id, peer_id, peer_id_len);
+	} else {
+		memcpy(operation->min_id, peer_id, peer_id_len);
+	}
+
+	/* calculate PWE */
+	if (operation->use_h2e) {
+		return cracen_wpa3_sae_calc_pwe_h2e(operation);
+	} else {
+		return cracen_wpa3_sae_calc_pwe_hnp(operation);
+	}
+}
+
+psa_status_t cracen_wpa3_sae_output(cracen_wpa3_sae_operation_t *operation,
+				    psa_pake_step_t step, uint8_t *output,
+				    size_t output_size, size_t *output_length)
+{
+	psa_status_t status = PSA_ERROR_CORRUPTION_DETECTED;
+
+	switch (step) {
+	case PSA_PAKE_STEP_COMMIT:
+		if (output_size < CRACEN_WPA3_SAE_COMMIT_SIZE) {
+			return PSA_ERROR_BUFFER_TOO_SMALL;
+		}
+
+		status = cracen_construct_commit_msg(operation);
+		if (status != PSA_SUCCESS) {
+			return status;
+		}
+		memcpy(output, operation->commit, CRACEN_WPA3_SAE_COMMIT_SIZE);
+		*output_length = CRACEN_WPA3_SAE_COMMIT_SIZE;
+		break;
+	case PSA_PAKE_STEP_CONFIRM:
+		if (output_size < CRACEN_WPA3_SAE_CONFIRM_SIZE) {
+			return PSA_ERROR_BUFFER_TOO_SMALL;
+		}
+
+		status = cracen_wpa3_sae_calc_keys(operation);
+		if (status) {
+			return status;
+		}
+
+		status = cracen_wpa3_sae_calc_verif_confirm(operation,
+							    operation->commit,
+							    operation->peer_commit,
+							    operation->send_confirm,
+							    output);
+		if (status) {
+			return status;
+		}
+		*output_length = CRACEN_WPA3_SAE_CONFIRM_SIZE;
+		break;
+	case PSA_PAKE_STEP_KEYID:
+		if (output_size < CRACEN_WPA3_SAE_PMKID_SIZE) {
+			return PSA_ERROR_BUFFER_TOO_SMALL;
+		}
+
+		memcpy(output, operation->pmkid, CRACEN_WPA3_SAE_PMKID_SIZE);
+		*output_length = CRACEN_WPA3_SAE_PMKID_SIZE;
+		break;
+	default:
+		return PSA_ERROR_INVALID_ARGUMENT;
+	}
+
+	return PSA_SUCCESS;
+}
+
+psa_status_t cracen_wpa3_sae_input(cracen_wpa3_sae_operation_t *operation,
+				   psa_pake_step_t step, const uint8_t *input,
+				   size_t input_length)
+{
+	psa_status_t status = PSA_ERROR_CORRUPTION_DETECTED;
+	uint8_t verify[CRACEN_WPA3_SAE_CONFIRM_SIZE];
+	uint16_t send_confirm;
+	int res;
+
+	switch (step) {
+	case PSA_PAKE_STEP_COMMIT:
+		if (input_length < CRACEN_WPA3_SAE_COMMIT_SIZE) {
+			return PSA_ERROR_INVALID_ARGUMENT;
+		}
+
+		status = cracen_check_commit_msg(operation, input);
+		if (status != PSA_SUCCESS) {
+			return status;
+		}
+		memcpy(operation->peer_commit, input, CRACEN_WPA3_SAE_COMMIT_SIZE);
+		break;
+	case PSA_PAKE_STEP_SALT:
+		/* rejected groups list */
+		status = cracen_wpa3_sae_setup_hmac(&operation->mac_op, input, input_length);
+		if (status) {
```

Comment:
nit bur for consistency lets stick to status != PSA_STATUS here nad bellow in the function.